src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingParameterSymbol.cs

```diff
@@ -118,7 +116,7 @@ internal override ImmutableArray<byte> MarshallingDescriptor
             }
         }
 
-        internal sealed override CSharpCompilation DeclaringCompilation // perf, not correctness
+        internal sealed override CSharpCompilation? DeclaringCompilation // perf, not correctness
```

Comment:
Not related to the PR: what in the world does this comment mean...? What perf?

Reply:
My estimation is that because `Symbol.DeclaringCompilation` effectively has this code: 

``` 
// omittted  
var sourceModuleSymbol = this.ContainingModule as SourceModuleSymbol;
return sourceModuleSymbol == null ? null : sourceModuleSymbol.DeclaringCompilation;
```

For `RetargetingParameterSymbol` the `SourceModuleSymbol` is always going to be `RetargetingModuleSymbol` which has 

``` 
        internal sealed override CSharpCompilation? DeclaringCompilation // perf, not correctness
        {
            get { return null; }
        }
```

That means the `override` doesn't change the output here but it is more performant because it avoids a `switch`, a type check and an additional virtual dispatch.

Reply:
Going to add a comment to this effect.